src/tests/prepare.spec.js

```diff
@@ -294,6 +308,246 @@ describe('prepare', () => {
     assert.equal(html, '<div>foo</div>', 'renders with correct html');
   });
 
+  const testPrepareComponent = async (componentCreator, expectedText) => {
+    const doAsyncSideEffect = sinon.spy(async () => {});
+    const prepareUsingProps = sinon.spy(async ({ text }) => {
+      await doAsyncSideEffect(text);
+    });
+    const ComponentTester = prepared(prepareUsingProps)(({ text }) => (
+      <div>{text}</div>
+    ));
+    const Component = componentCreator(ComponentTester);
+    await prepare(<Component />);
+
+    assert(
+      prepareUsingProps.calledOnce,
+      'prepareUsingProps has been called once',
+    );
+    assert(
+      prepareUsingProps.calledWith({ text: expectedText }),
+      'prepareUsingProps has been called with correct props',
+    );
+    assert(
+      doAsyncSideEffect.calledOnce,
+      'doAsyncSideEffect has been called once',
+    );
+
+    // if it doesn't render to correct html, the test is likely set up wrong
+    const html = renderToStaticMarkup(<Component />);
+    assert.equal(
+      html,
+      `<div>${expectedText}</div>`,
+      'renders with correct html',
+    );
+  };
+
+  it('Should support useState() hook', async () => {
+    await testPrepareComponent(
+      // eslint-disable-next-line react/display-name
```

Comment:
pls disable this globally for the file if you want to do it this way

Reply:
Yeah, i probably should! I could probably disable it for the whole project. React components are only created in tests, so it's not really relevant